hazelcast/src/test/java/com/hazelcast/partition/PartitionMigrationListenerTest.java

```diff
@@ -92,8 +98,7 @@ public void testMigrationStats_whenMigrationProcessRestarts() {
         config.setProperty(ClusterProperty.PARTITION_COUNT.getName(), String.valueOf(partitionCount));
         config.setProperty(ClusterProperty.PARTITION_MAX_PARALLEL_MIGRATIONS.getName(), String.valueOf(1));
 
-        HazelcastInstance hz1 = factory.newHazelcastInstance(config);
-        warmUpPartitions(hz1);
+        HazelcastInstance hz1 = createPausedMigrationCluster(factory, Optional.of(config));
```

Comment:
This test previously set the cluster state to `NO_MIGRATION` after conducting other operations - have you made sure the moving of this state change has no negative impact on this test?

Reply:
The test still passes and when writing my test, the order of setting up the partitions, toggling migration state and adding migration listeners had no impact until another cluster was brought into the equation. So I'm pretty confident.